[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion

ZhiweiYan-96 opened the pull request:
# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118064
* #117824
* #117529
* #117512



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

ZhiweiYan-96 committed:
Intel GPU oneDNN upstreaming: Conv pointwise fusion

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"




[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

xiaolil1 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

xiaolil1 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

EikanWang reviewed (commented):
Please add test case.

leizhenyuan committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

jgong5 reviewed (commented):
How are you going to test your code?

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

ZhiweiYan-96 committed:
Update on "[WIP]Intel GPU oneDNN upstreaming: Conv pointwise fusion"

# Motivation

This PR is a part of RFC #114848,. This PR would depends on #117529  and its further predecessor PRs. 

This PR is used to add pointwise fusion support for convolution. The conv-pointwise fusion is expected to be used in graph mode like `torch.compile`.  The fusion utils is alread contained in #117824. With `conv-pointwise` and `linear-pointwise` fusion, we may be able to inference lost of models in benchmark like TorchBench.




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]
